lib/buildFactory.js

```diff
@@ -236,6 +245,31 @@ class BuildFactory extends BaseFactory {
             });
     }
 
+    /**
+     * Get a build based on id
+     * @method get
+     * @param  {Mixed}   config    The configuration from which an id is generated or the actual id
+     * @return {Promise}           Resolve build model after merging with step models
+     */
+    get(config) {
+        return super.get(config)
+            .then(build => build.getStepsModel()
+                .then((stepsModel) => {
+                    build.steps = build.steps.map((step) => {
+                        const stepModel = stepsModel.find(s => s.name === step.name);
```

Comment:
If `stepModel exists can't we do `build.steps = stepModel` ?

Reply:
No, need to get every single model and do `model.toJson()` before returning it otherwise it will contain functions and other stuff. I added this in the latest commit.